pkg/server/plugin/datastore/sql/list_attested_nodes_query_test.go

```diff
@@ -0,0 +1,2500 @@
+	NULL AS selector_value
+FROM filtered_nodes
+WHERE id IN (
+	SELECT id FROM (
+		SELECT id 
+		FROM filtered_nodes ORDER BY id ASC LIMIT 1
+	) workaround_for_mysql_subquery_limit
+)
+`},
+		{
+			dialect:     "mysql",
+			supportsCTE: true,
+			by:          []string{"attestation-type", "banned-true", "selector-exact-many", "expires-before"},
+			paged:       "with-token",
+			query: `
+WITH filtered_nodes AS (
+	SELECT * FROM attested_node_entries WHERE true
+		AND id > ?		AND expires_at < ?
+		AND data_type = ?
+		AND serial_number = ''
+), filtered_nodes_and_selectors AS (
+	    SELECT
+	        filtered_nodes.*, nr.type AS selector_type, nr.value AS selector_value
+	    FROM
+			filtered_nodes
+	    LEFT JOIN
+	 	    node_resolver_map_entries nr       
+	    ON
+	        nr.spiffe_id=filtered_nodes.spiffe_id
+	)
+
+SELECT 
+	id as e_id,
+	spiffe_id,
+	data_type,
+	serial_number,
+	expires_at,
+	new_serial_number,
+	new_expires_at,
+	selector_type,
+	selector_value 
+	  
+FROM filtered_nodes_and_selectors
+WHERE id IN (
+	SELECT id FROM (
+	SELECT DISTINCT id FROM (
+		(SELECT id FROM filtered_nodes_and_selectors WHERE selector_type = ? AND selector_value = ?) c_0
+		INNER JOIN
+		(SELECT id FROM filtered_nodes_and_selectors WHERE selector_type = ? AND selector_value = ?) c_1
+		USING(id)
+
+	)  ORDER BY id ASC LIMIT 1
+	) workaround_for_mysql_subquery_limit
+)
+`},
+		{
+			dialect:     "mysql",
+			supportsCTE: true,
+			by:          []string{"attestation-type", "banned-true", "selector-exact-many", "expires-before"},
+			paged:       "no-token",
+			query: `
+WITH filtered_nodes AS (
+	SELECT * FROM attested_node_entries WHERE true
+		AND expires_at < ?
+		AND data_type = ?
+		AND serial_number = ''
+), filtered_nodes_and_selectors AS (
+	    SELECT
+	        filtered_nodes.*, nr.type AS selector_type, nr.value AS selector_value
+	    FROM
+			filtered_nodes
+	    LEFT JOIN
+	 	    node_resolver_map_entries nr       
+	    ON
+	        nr.spiffe_id=filtered_nodes.spiffe_id
+	)
+
+SELECT 
+	id as e_id,
+	spiffe_id,
+	data_type,
+	serial_number,
+	expires_at,
+	new_serial_number,
+	new_expires_at,
+	selector_type,
+	selector_value 
+	  
+FROM filtered_nodes_and_selectors
+WHERE id IN (
+	SELECT id FROM (
+	SELECT DISTINCT id FROM (
+		(SELECT id FROM filtered_nodes_and_selectors WHERE selector_type = ? AND selector_value = ?) c_0
+		INNER JOIN
+		(SELECT id FROM filtered_nodes_and_selectors WHERE selector_type = ? AND selector_value = ?) c_1
+		USING(id)
+
+	)  ORDER BY id ASC LIMIT 1
+	) workaround_for_mysql_subquery_limit
+)
+`},
+	} {
+		tt := tt
+		name := tt.dialect + "-list-"
+		if len(tt.by) == 0 {
+
+		} else {
+			name += "by-" + strings.Join(tt.by, "-")
+		}
+		if tt.paged != "" {
+			name += "-paged-" + tt.paged
+		}
+		if tt.supportsCTE {
+			name += "-cte"
+		}
+		expiresBefore := time.Now().Unix()
+		t.Run(name, func(t *testing.T) {
+			req := new(datastore.ListAttestedNodesRequest)
+			switch tt.paged {
```

Comment:
I think that this switch could be more clear if it's done through a custom type.